src/pages/blocks/[blockId]/index.tsx

```diff
@@ -0,0 +1,221 @@
+import { Transaction } from '@defichain/whale-api-client/dist/api/transactions'
+
+import { Breadcrumb } from '@components/commons/Breadcrumb'
+import { OverflowTable } from '@components/commons/OverflowTable'
+import { CursorPage, CursorPagination } from '@components/commons/CursorPagination'
+import { getWhaleApiClient } from '@contexts/WhaleContext'
+import { IoCopyOutline } from 'react-icons/io5'
+
+import { fromUnixTime, format } from 'date-fns'
+
+interface BlockDetailsPageProps {
+  block: Block
+  confirmations: number
+  transactions: {
+    items: Transaction[]
+    pages: CursorPage[]
+  }
+}
+
+function CopyButton ({ text }: { text: string }): JSX.Element {
+  const [open, setOpen] = useState<Boolean>(false)
+
+  async function copy (): Promise<void> {
+    await navigator.clipboard.writeText(text)
+    setOpen(true)
+
+    setTimeout(() => {
+      setOpen(false)
+    }, 500)
+  }
+
+  return (
+    <div className='ml-1'>
+      <button className='cursor-pointer outline-none p-1 bg-gray-100 rounded shadow-sm' onClick={copy}>
+        <IoCopyOutline className='h-5 w-5 text-gray-500' />
+      </button>
+      {
+        open === true
+          ? (
+            <span className='bg-gray-100 p-1 rounded absolute mr-4'>
+              Copied!
+            </span>
+          )
+          : null
+      }
+    </div>
+  )
+}
+
+function BlockDetail (props: { label: string, text: string, children?: ReactNode, testId?: string }): JSX.Element {
+  const { label, text, children, testId } = props
+  return (
+    <div className='px-6 py-4 flex justify-between'>
+      <span className='w-1/3 flex-shrink-0'>
+        {label}
+      </span>
+      {
+        (children !== undefined)
+          ? children
+          : (
+            <span className='flex-grow' data-testid={testId}>
+              {text}
+            </span>
+          )
+      }
+    </div>
+
+  )
+}
+
+BlockDetail.defaultProps = {
+  text: '',
+  label: ''
+}
+
+export default function BlockDetails ({ block, confirmations, transactions }: InferGetServerSidePropsType<typeof getServerSideProps>): JSX.Element {
+  const leftBlockDetails = [
+    {
+      label: 'Block Reward:',
+      text: 'xxxx DFI'
+    },
+    {
+      label: 'Height:',
+      text: `${block.height}`,
+      testId: 'block-detail-height'
+
+    },
+    {
+      label: 'Transactions:',
+      text: `${block.transactionCount}`,
+      testId: 'block-detail-transactions'
+    },
+    {
+      abel: 'Timestamp:',
+      text: `${format(fromUnixTime(block.medianTime), 'PPpp')}`,
+      testId: 'block-detail-timestamp'
+    },
+    {
+      label: 'Confirmations:',
+      text: `${confirmations}`,
+      testId: 'block-detail-confirmations'
+    },
+    {
+      label: 'Merkle Root:',
+      children:
+  <>
+    <span className='flex-grow break-all' data-testid='block-detail-merkle-root'>{block.merkleroot}</span>
+    <CopyButton text={block.merkleroot} />
+  </>
+    }
+  ]
```

Comment:
Turn this into a natural HTML hierarchy, don't do any functional transformation on static data. It just makes code harder to maintain, and harder to trace.

```html
<BlockDetail/>
<BlockDetail/>
<BlockDetail/>
<BlockDetail/>
```

Reply:
Changed.